tools/lab_dispatch/generic_source.yaml

```diff
@@ -0,0 +1,78 @@
+job_queue: $INPUT_QUEUE
+global_timeout: 3600
+output_timeout: 1800
+provision_data:
+  $INPUT_DATA_SOURCE
+test_data:
+  attachments:
+    - local: "tools/lab_dispatch/resources/manifest.conf"
+      agent: "manifest.conf"
+    - local: "tools/lab_dispatch/resources/input.conf"
+      agent: "input.conf"
+    - local: "tools/lab_dispatch/resources/service_file.service"
+      agent: "service_file.service"
+    - local: "tools/lab_dispatch/build_install_deb.py"
+      agent: "build_install_deb.py"
+  test_cmds: |
+    #!/usr/bin/env bash
+
+    set -x
+
+    # input arguments
+    CHECKBOX_REVISION=$INPUT_CHECKBOX_REVISION
+    RESOURCES_PATH="attachments/test"
+
+    TOOLS_PATH=tools
+
+    # retrieve all scripts/tools necessary from a repo
+    curl -Ls -o install_tools.sh https://raw.githubusercontent.com/canonical/hwcert-jenkins-tools/main/install_tools.sh
+    source install_tools.sh $TOOLS_PATH
+
+    # ensure device is available before continuing
+    wait_for_ssh
+
+    _run sudo add-apt-repository ppa:checkbox-dev/edge
+    _run install_packages git python3 python3-pip dpkg-dev
+
+    wait_for_ssh
+
+    _put $RESOURCES_PATH/build_install_deb.py :
+    _run git clone https://github.com/canonical/checkbox.git
+    _run git -C checkbox checkout $CHECKBOX_REVISION
+    _run python3 build_install_deb.py --clean checkbox/checkbox-ng \
+        checkbox/checkbox-support checkbox/providers/resource \
+        checkbox/providers/base checkbox/providers/sru
+    _run sudo systemctl restart checkbox-ng.service
+
+    git clone https://github.com/canonical/checkbox.git
+    git -C checkbox checkout $CHECKBOX_REVISION
+
+    python3 -m venv --system-site-packages checkbox_venv
+    . checkbox_venv/bin/activate
+    pip install --upgrade pip setuptools
```

Comment:
This is really interesting. After some [conversations I had with Paul on this PR](https://github.com/canonical/hwcert-jenkins-tools/pull/36) the installer actually installs `pipx` and `python3-venv` as deb packages, i.e. not through `pip`. We might discuss which of the two approaches is best but what I am essentially saying is that you could probably use whatever the installer has already installed.

And I am wondering whether you can directly call `pipx install checkbox/checkbox-ng` without explicitly activating the virtual environment (as `pipx` will handle that for you). 

**Update**: Actually, `install_checkbox_agent_source` [does exactly that](https://github.com/canonical/hwcert-jenkins-tools/blob/61e32fe28d0060fabfe3621af2f43713d6f5398e/scriptlets/install_checkbox_agent_source#L48). Would you be able to use that scriptlet directly?

Reply:
no because it checks out to a revision, so I cant use the scriplet. I can use pipx, I used normal pip and a venv because I'm just more used to that. I can try this.